formatencoder.go

```diff
@@ -29,7 +29,14 @@ func init() {
 	caddy.RegisterModule(FormattedEncoder{})
 }
 
-const commonLogFormat = `{common_log}`
+const (
+	// commonLogFormat is the common log format. https://en.wikipedia.org/wiki/Common_Log_Format
+	commonLogEmptyValue = "-"
+	commonLogFormat     = `{request>remote>host} ` + commonLogEmptyValue + ` {request>user_id} [{ts}] "{request>method} {request>uri} {request>proto}" {status} {size}`
```

Comment:
@francislavoie, the `{request>remote>host}` field does not exist in the current JSON log. It has the remote_addr in hostport.

Reply:
I think you can just use `{host}`, no?

https://github.com/caddyserver/caddy/blob/d4b2f1bcee2190d13bc96aa8da5ea530bd6b9a22/modules/caddyhttp/marshalers.go#L32

Reply:
Oh wait this is the client's IP. Right. Uh...

Reply:
Is there a way for this templating to invoke go template functions? Or is that too expensive? Like split on `:` type of deal. I don't think we really want to change Caddy's logging just to conform to common log here 😬

Reply:
We should not be evaluating templates at every single log emission if possible.

Reply:
I wonder if this package could come in handy 🤔 

https://pkg.go.dev/rsc.io/tmplfunc

Reply:
Alternate idea @Mohammed90 we could implement a `regexp` filter possibly? https://caddyserver.com/docs/caddyfile/directives/log#filter it would be a replacement, so we'd tell people to add a filter to match on `:[0-9]+` and replace it with `""`

Reply:
Is there room to argue for the presence of `remote_host`? It can be beneficial to look at and analyze, for example for fail2ban to pick up and block malicious actors.

Reply:
We already have `remote_addr`, we don't want to log both. `remote_host` is lossy compared to `remote_addr` since it would just be with the port dropped.

Reply:
I'll bite the bullet and use `remote_addr` even though it isn't exactly matching the spec

Reply:
Ultimately for fail2ban people and such, they should be able to configure it to ignore the port anyways. So it should be fine.